packit/cli/validate_config.py

```diff
@@ -27,18 +30,65 @@
     is_flag=True,
     help="Do not make remote API calls requiring network access.",
 )
+@click.option(
+    "-c",
+    "--config",
+    type=click.Path(exists=True),
+    help="Path to a specific Packit configuration file.",
+)
 @cover_packit_exception
-def validate_config(path_or_url: LocalProject, offline: bool):
+def validate_config(path_or_url: LocalProject, offline: bool, config: str = None):
     """
     Validate PackageConfig.
 
     \b
     - checks missing values
     - checks incorrect types
-    - checks whether monitoring is enabled if 'pull_from_upstream` is used
+    - checks whether monitoring is enabled if 'pull_from_upstream' is used
 
-    PATH_OR_URL argument is a local path or a URL to a git repository with packit configuration file
+    PATH_OR_URL argument is a local path or a URL to a git repository with a packit configuration file.
     """
-    output = PackitAPI.validate_package_config(path_or_url.working_dir, offline)
-    logger.info(output)
+    try:
+        logger.info(
+            (
+                f"Using configuration file: {config}"
+                if config
+                else "No custom config provided."
+            ),
+        )
+
+        config_path = (
+            Path(config)
+            if config
+            else find_packit_yaml(path_or_url.working_dir, try_local_dir_last=True)
+        )
+
+        logger.info(f"Validating config file: {config_path}")
+
+        if not config_path.exists():
+            raise FileNotFoundError(f"Configuration file not found: {config_path}")
+
+        try:
+            config_content = load_packit_yaml(config_path)
+        except yaml.YAMLError as e:
+            logger.error(f"Failed to parse YAML file: {e}")
+            raise
+
+        validator = PackageConfigValidator(
+            config_path,
+            config_content,
+            path_or_url.working_dir,
+            offline,
+        )
+
+        output = validator.validate()
+        logger.info(output)
+
+    except FileNotFoundError as e:
+        logger.error(f"Error: {e}")
+    except yaml.YAMLError as e:
+        logger.error(f"YAML parsing error: {e}")
+    except Exception as e:
+        logger.error(f"Unexpected error: {e}")
```

Comment:
It would be nice if we could resolve these error states and fail soo. (It helps code readability if we don't go deeper (~right) into code structure.)